### PR TITLE
Bump Node to 22

### PR DESCRIPTION
**Pull request by ScottBrenner**
Recreating https://github.com/openai/main-branch-check-action/pull/12 here

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     - name: setup node
       uses: actions/setup-node@v4
       with:
-        node-version: '21.x'
```
Comment (ScottBrenner) [inline .github/workflows/build.yaml:18]: (Noticed the `.x` does not resolve https://github.com/jaymzh/main-branch-check-action/actions/runs/9169005598/job/25208725652#step:3:8)

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
Thanks!

**Comment by jaymzh** [comment]
@ScottBrenner - looks like this broke things:

```
Download action repository 'jaymzh/main-branch-check-action@main' (SHA:2fa5938f39733174b68c0e6b4a44ae9f745f451a)
Error: System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter ''using: node22' is not supported, use 'docker', 'node12', 'node16' or 'node20' instead.')
```

**Comment by ScottBrenner** [comment]
Odd, does https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing-for-javascript-actions not support 22? https://github.com/actions/setup-node does as of  https://github.com/actions/node-versions/releases/tag/22.0.0-8879734543

**Comment by jaymzh** [comment]
`setup-node@v4` is what complained...

**Comment by jaymzh** [comment]
BTW I cleaned up the setup instructions and submitted it to Marketplace.

**Comment by ScottBrenner** [comment]
Ah sorry, was on mobile when I sent that and did not type it out entirely-

setup-node@v4 works perfectly fine with Node 22, as logged by the passing job in this PR - https://github.com/jaymzh/main-branch-check-action/pull/1/checks#step:3:9

That error is logged by the "runner", on which this action is executed - commented regarding this on https://github.com/actions/runner/discussions/2704#discussioncomment-9515846

**Comment by jaymzh** [comment]
Good call. Though it still meant referring to 20. Thanks for reporting that I'll keep an eye on it.

**Pull request merged**